sentry-rails/lib/sentry/rails/tracing/active_record_subscriber.rb

```diff
@@ -31,6 +31,10 @@ def self.subscribe!
                             connection.pool.db_config.configuration_hash
                           elsif connection.pool.respond_to?(:spec)
                             connection.pool.spec.config
+                          # CockroachDB pool shows up as NullPool, but we can grab
+                          # it's configuration from the instance variable.
+                          elsif connection.instance_variable_defined?(:@config)
```

Comment:
can you point in the rails/cockroach adapter source where this `:config` var is defined?

Reply:
I have a sneaky suspicion that `@config` on the connection is available in any connection, not just CockroachDB: https://github.com/rails/rails/blob/9c22f35440ab85718ebf48e26b8944032c737193/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb#L135C36-L135C36

That's what makes this code more or less sane to me — if all else fails, the config should be a hash on the connection object, whatever it is.

Reply:
yea but that's an instance variable, not part of the official class api. There was a reason I used the pool/db_config when I implemented this because that's the standard way in rails now.